### DISCUSSION TITLE
Icon Request: icon-rename

### DISCUSSION THREAD

**Discussion by rsteuber** [FortAwesome/Font-Awesome#21150] [Icon Requests]
Hello,

First of all i would like to thank you for the awesome icons you have been creating.
I very much enjoy them.

Ive been searching for an icon for my filemanager that represents the (rename action) for a file.

Hope you can help me with that.

Best regards,
Rob Steuber

**Comment by brendanheyu** [comment]
+1

**Comment by RobIverson** [comment]
+1

**Comment by einsitang** [comment]
same here
+1

**Comment by M-a-r-t-i-j-n** [comment]
+1

I use fa-edit (pencil on square) for editing an object, so I need another kind of icon for renaming.

For fa-rename I suggest a typical text-editor kind of cursor, optionally with a rectangle suggesting an html text input (and hence a short text).

![text-cursor](https://cloud.githubusercontent.com/assets/11576937/6758865/c63631b8-cf3b-11e4-8923-7d35e5091d62.gif)

![text-cursor2](https://cloud.githubusercontent.com/assets/11576937/6758855/b87abab2-cf3b-11e4-88e1-5a7a1d055ac2.gif)

It could perhaps even be called fa-cursor or fa-text-cursor with fa-rename as an alias.

**Comment by mhdhadjar** [comment]
+1

**Comment by leonkladnitsky** [comment]
+1

**Comment by gurisko** [comment]
+1

**Comment by dennis-g** [comment]
+1

**Comment by voxtobox** [comment]
+1 
![editing-rename-icon](https://cloud.githubusercontent.com/assets/13472105/13949083/24d393f2-f02c-11e5-9a00-2f4014c5d3f1.png)
![1458661543_rename](https://cloud.githubusercontent.com/assets/13472105/13949084/24d8874a-f02c-11e5-99c7-b920f28115f8.png)

**Comment by jackbentley** [comment]
For anyone stumbling upon this, you could use [fa-i-cursor](http://fontawesome.io/icon/i-cursor/).

However, it would be nice to see the version with the box included at a later date, also.

(Maybe also add the tag "text" to the `fa-i-cursor` icon for the search box in docs?)

**Comment by leonkladnitsky** [comment]
You can make your own version with a box:

<i class="fa fa-border" aria-hidden="true" style="border-color: black;
padding: 0.1em 0.5em 0.1em 0.2em;">
    <i class="fa fa-i-cursor" aria-hidden="true"></i>
</i>

I prefer fa-pencil though.

**Comment by brianmcmichael** [comment]
+1

**Comment by dibley1973** [comment]
+1

**Comment by insign** [comment]
+1

**Comment by DaleyKD** [comment]
7 years later... can we get this?

**Comment by HobieCat** [comment]
+1

**Comment by ManitaJ** [comment]
There are object in applications that are created and edited on one place and can be renamed in another. Can we please have a rename icon for that action.

**Comment by drewTheNerd** [comment]
+1

**Comment by bradar93** [comment]
+1

**Comment by sensibleworld** [comment]
This is a quick update to let folks know that we’re refining our icon request workflow behind the scenes, and as part of that we’re going through all open requests and filing them into several new buckets to be addressed.

In this case we’ve added this issue to our new list to be created! We can’t promise any firm timelines right now, but this will help us keep track of it.

We’ll update this issue when the icon is released into the wild 🙂
